Cargo.toml

```diff
@@ -82,6 +82,7 @@ serde_repr = "0.1.16"
 serde_with = "3.4.0"
 tempfile = "3.8"
 tokio = { version = "1", features = ["macros"] }
+tokio-shared-rt = "0.1.0"
```

Comment:
Hi, I'm a bit nervous about introducing `tokio-shared-rt`. It's inactive and rarely used. Additionally, it might unexpectedly lead developers to share tests, which goes against our usual rust test pattern.

If we do want this, I will encourage to write our own test harness.

Reply:
ok, i will use our tools to share tokio runtime between test case. But a share tokio runtime is indeed needed.
Because static shared resources maybe contains some resource rely on tokio runtime, e.g. tokio Mutex .So if one of those test case exit, tokio runtime is droped, the static shared resources throw `A Tokio 1.x context was found, but it is being shutdown.`

So here I want to rewrite our own macro, or do something like this? 
```
pub fn test<F: std::future::Future>(f: F) -> F::Output {
    rt().handle().enter(|| futures::executor::block_on(f))
}

#[test]
fn a_test() {
    test(async {
        println!(" this is async");
    });
}
```

Reply:
> So here I want to rewrite our own macro, or do something like this?

This leads to the solution I proposed in https://github.com/apache/iceberg-rust/issues/425: We need our own test harness if we want `cargo test` still works.

Reply:
You are right, but here single test also is needed.